Deployment types guidance for writers (#6745)
elastic/docs-content: contribute-docs/_snippets/self-managed-naming.md

:::{warning}

> Some people refer to everything not hosted on {{ecloud}} as "self-managed" (grouping ECK, ECE, and self-managed together). **Don't use this grouping in our docs.** ECK, ECE, and self-managed differ in meaningful ways that the grouping obscures. Always name the specific deployment type.

Reviewer: Perfect example where passive voice helps diffuse finger-pointing 😅 

```suggestion
Sometimes, everything that is not hosted on {{ecloud}} might incorrectly be referred to as "self-managed" (grouping ECK, ECE, and self-managed together). **Don't use this grouping in our docs.** ECK, ECE, and self-managed differ in meaningful ways that the grouping obscures. Always name the specific deployment type.
```

or 

```suggestion
The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and tying them together obscures those distinctions. Always refer to each deployment type by name.
```

> The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and tying them together obscures those distinctions. Always refer to each deployment type by name.

Reviewer 2: ```suggestion
The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. But ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and conflating them obscures those distinctions. Always refer to each deployment type by name.
```

Reviewer 2: Very minor nits. I guess `conflating` is not super readable. But "tying them together" seems imprecise somehow.

> The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and tying them together obscures those distinctions. Always refer to each deployment type by name.

Author: ```suggestion
The term "self-managed" is sometimes used as a catch-all for anything not hosted on {{ecloud}}, including ECK and ECE. ECK, ECE, and self-managed are distinct deployment types with meaningful differences, and treating them as a unit obscures those distinctions. Always refer to each deployment type by name.
```
